### PR TITLE
feat: enable elliptic curve es256 for JWS

### LINKED ISSUE #203: The algorithm for JWS is hardcoded to RS256
According to the JWS standard, more algorithms are allowed: https://datatracker.ietf.org/doc/html/rfc7518#section-3.1
Elliptic curve algorithms have better performance and might be preferred in cases where the performance matters.

As per [NIST Special Publication 800-57 Part 1](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r5.pdf), page 54, table 2 the algorithm ES256 has a cryptographic strength of 128 bits, which is equivalent to RS256 with 3072 bit key, which satisfies the current Mojaloop requirements for [signature generation](https://docs.mojaloop.io/api/fspiop/v1.1/signature.html#generating-a-signature).

### PR DESCRIPTION
**Pull request by kalinkrustev**
Closes #203 
- During signing the algorithm is determined based on the contents of the key file
- During verification both RS256 and ES256 are allowed.
- Added a performance test to illustrate the impact.
  The result is available [here](https://app.circleci.com/pipelines/github/mojaloop/sdk-standard-components/931/workflows/8f5977c2-5d82-4ce6-b9bc-3964c19ddc72/jobs/3972?invite=true#step-107-11001_33) and shows much better performance when using ES256:
  image

### CODE REVIEW

**Review by elnyry-sam-k — DISMISSED** [review]

**Comment by elnyry-sam-k** [comment]
We have option for both so I'm good for now. We need to continue to use the RS256 on prod / official deployments until the new algorithm is approved by CCB / FSPIOP SIG.

**Comment by kalinkrustev** [comment]
@elnyry-sam-k , this needs an approval again, as there were merge conflicts and dependency updates

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**
